lib/web/device_trust.go

```diff
@@ -76,7 +81,39 @@ func (h *Handler) deviceWebConfirm(w http.ResponseWriter, r *http.Request, _ htt
 
 	// Always redirect back to the dashboard, regardless of outcome.
 	app.SetRedirectPageHeaders(w.Header(), "" /* nonce */)
-	http.Redirect(w, r, "/web", http.StatusSeeOther)
+
+	redirectTo, err := h.getRedirectURL(redirectURI)
+	if err != nil {
+		h.log.Debugf("unable to parse redirectURI %s", redirectURI)
```

Comment:
Use structured logging:

```suggestion
		h.log.
			WithError(err).
			WithField("redirect_uri", redirectURI).
			Debug("Unable to parse redirectURI")
```